mod_papers/helper.php

```diff
@@ -20,6 +21,29 @@ class ModPapersHelper
      *
      * @access public
      */
+    public static function showPapers($orcids){
+
+        $config = include('config.php');
+	$lastRunLog =  dirname(__FILE__) . '/lastrun.log';
+	$data_file = dirname(__FILE__) . '/pubs_cache_file.html';
+
+	$lastRun = file_get_contents($lastRunLog);
+
+	if (time() - (int)$lastRun >= $config['update_time']) {
+	    //its been more than a day so run our external file
+	    $papers = self::getPapers($orcids);
+	    //update lastrun.log with current time
+	    file_put_contents($lastRunLog, time());
+	    $myfile = fopen($data_file, "w") or die("Unable to open file!");
+	    fwrite($myfile, $papers);
+	    fclose($myfile);
+	} else {
+	    $papers = file_get_contents($data_file);	
+	}
+
+	return $papers;
+    }
+
```

Comment:
Looks good. Shouldn't be too hard to swap to a passed in `$config` if you agree with that.

`getPapers` could now be private.